src/assets/JourneyWithGo-ABlogSeries/goControlFlow.md

```diff
@@ -0,0 +1,378 @@
+
+```go
+if condition {
+    // code to execute if the condition is true
+}
+```
+
+Example:
+
+![image](/src/assets/JourneyWithGo-ABlogSeries/images/if-example.png "If Example")
+
+Output:
+
+```go
+You are eligible to receive a Citizenship
+```
+
+##### _b. if-else_
+
+When you want to execute codes either for true or false case then `if-else` can be used. `if-else` executes the code inside of the `if` block if the condition is true otherwise, it executes the code inside of the `else` block. The syntax of `if-else` is as follows.
+
+```go
+if condition {
+    // code to execute if the condition is true
+} else {
+    // code to execute if the condition is false
+}
+```
+
+Example:
+
+![image](/src/assets/JourneyWithGo-ABlogSeries/images/if-else-example.png "If-Else Example")
+
+Output:
+
+```go
+You are eligible to vote.
+```
+
+##### _c. nested if-else_
+
+If the if-else statement is inside another if-else statement, it is called a nested if-else statement. We can use nested if-else statements to check another conditions if previous condition is met. We have different variation of nested if-else statements. We will discuss only one of them.
+The syntax of `nested if-else` is as follows.
+
+```go
+if condition {
+    // code to execute if the condition is true
+    if condition {
+        // code to execute if the condition is true
+    } else {
+        // code to execute if the condition is false
+    }
+} else {
+    if condition {
+        // code to execute if the condition is true
+    } else {
+        // code to execute if the condition is false
+    }
+}
+```
+
+Example:
+
+![image](/src/assets/JourneyWithGo-ABlogSeries/images/nested-if-example.png "Nested If-Else Example")
+
+Output:
+
+```go
+The greatest number is: 566
+```
+
+##### _d. else-if ladder_
+
+Else-if ladder is a series of if-else statements. It is used when we have multiple conditions to check, when the previous condition is not met. The syntax of `else-if ladder` is as follows.
+
+```go
+if condition {
+    // code to execute if the condition is true
+} else if condition {
+    // code to execute if the condition is true
+} else if condition {
+    // code to execute if the condition is true
+} else {
+    // code to execute if none of the conditions are true
+}
+```
+
+Example:
+
+![image](/src/assets/JourneyWithGo-ABlogSeries/images/else-if-ladder-example.png "Else-If Ladder Example")
+
+Output:
+
+```go
+It is Negative number
+```
+
+### 2. Switch
+
+`switch` allows to execute one block of codes among several cases based on the value of a given expression. It can be used as alternative to the `if-else` statement, when you have lots of condition to be checked. If none of the cases match with the expression, the code inside of the `default` block gets executed. The syntax of `switch` is as follows.
```

Comment:
```suggestion
`switch` allows us to execute one block of code among several cases based on the value of a given expression. It can be used as an alternative to the `if-else` statement when you have lots of conditions to be checked. If none of the cases match with the expression, the code inside of the `default` block gets executed. The syntax of `switch` is as follows.
```